File: APM/Java/Interpreter/src/ro/sandorrobertk94/domain/PrintStatement.java
package ro.sandorrobertk94.domain;

/**
 * Created by Robert on 10/11/2015.
 */
public class PrintStatement implements IStatement {
    private Expression exp;

    public PrintStatement() {
        this.exp = null;
    }

    public PrintStatement(Expression exp) {
        this.exp = exp;
    }

    public Expression getExp() {
        return exp;
    }

    public void setExp(Expression exp) {
        this.exp = exp;
    }

    @Override
    public String toString() {
        return "print ( " + exp.toString() + " )";
    }
}

File: APM/CSharp/Interpreter/Interpreter/domain/PrintStatement.cs
namespace Interpreter.domain
{
    public class PrintStatement : IStatement
    {
        private Expression _exp;

        public PrintStatement()
        {
            _exp = null;
        }

        public PrintStatement(Expression exp)
        {
            _exp = exp;
        }

        public Expression Exp
        {
            get { return _exp; }
            set { _exp = value; }
        }

        public override string ToString()
        {
            return "print ( " + _exp.ToString() + " )";
        }
    }
}